unit_test/batched/Test_Batched_TeamVectorSolveUTV2.hpp

```diff
@@ -0,0 +1,250 @@
+      std::string name = name_region + name_value_type;
+      Kokkos::Profiling::pushRegion( name.c_str() );
+
+      const int league_size = _a.extent(0);
+      Kokkos::TeamPolicy<DeviceType> policy(league_size, Kokkos::AUTO);
+
+      Kokkos::parallel_for(name.c_str(), policy, *this);
+      Kokkos::Profiling::popRegion(); 
+    }
+  };
+
+  template<typename DeviceType,
+           typename MatrixViewType,
+           typename VectorViewType,
+	   typename PivViewType,
+           typename WorkViewType,
+           typename AlgoTagType>
+  void impl_test_batched_solve_utv2(const int N, const int BlkSize) {
+    typedef typename MatrixViewType::non_const_value_type value_type;
+    typedef Kokkos::Details::ArithTraits<value_type> ats;
+    //const value_type one(1);
+    /// randomized input testing views
+    MatrixViewType r("r", N, BlkSize, 3);    
+    MatrixViewType a("a", N, BlkSize, BlkSize);
+    MatrixViewType acopy("copy", N, BlkSize, BlkSize);    
+    MatrixViewType u("u", N, BlkSize, BlkSize);
+    MatrixViewType v("v", N, BlkSize, BlkSize);
+    PivViewType    p("p", N, BlkSize);
+    VectorViewType x("x", N, BlkSize, 2);
+    VectorViewType b("b", N, BlkSize, 2);
+    WorkViewType   w("w", N, 3*BlkSize*2);
+
+    Kokkos::fence();
+
+    Kokkos::Random_XorShift64_Pool<typename DeviceType::execution_space> random(13718);
+    if (BlkSize <= 3) 
+      Kokkos::fill_random(a, random, value_type(1.0));
+    else 
+      Kokkos::fill_random(r, random, value_type(1.0));
+    
+    Kokkos::fence();
+
+    Functor_TestBatchedTeamVectorSolveUTV2
+      <DeviceType,MatrixViewType,VectorViewType,
+       PivViewType,WorkViewType,AlgoTagType>(r,a,acopy,u,v,p,x,b,w).run();
+
+    Kokkos::fence();
+
+    /// for comparison send it to host
+    auto a_host = Kokkos::create_mirror_view(acopy);
+    auto x_host = Kokkos::create_mirror_view(x);
+    auto b_host = Kokkos::create_mirror_view(b);
+    auto w_host = Kokkos::create_mirror_view(w);
+
+    Kokkos::deep_copy(a_host, acopy);    
+    Kokkos::deep_copy(x_host, x);
+    Kokkos::deep_copy(b_host, b);
+
+    /// this is least square; we cannot expect high
+    typedef typename ats::mag_type mag_type;
+    const mag_type eps = 1e3 * ats::epsilon();
+
+    for (int k=0;k<N;++k) {
+      mag_type residual(0), norm(0);
+
+      for (int l=0;l<2;++l) {
+	for (int i=0;i<BlkSize;++i) {
+	  value_type tmp(0);
+	  for (int j=0;j<BlkSize;++j) {	
+	    tmp += a_host(k,i,j)*x_host(k,j,l);
+	  }
+	  w_host(k,i) = tmp - b_host(k,i,l);
+	}
+	for (int i=0;i<BlkSize;++i) {
+	  value_type tmp(0);
+	  for (int j=0;j<BlkSize;++j) {	
+	    tmp += a_host(k,j,i)*w_host(k,j);
+	  }
+	  residual += ats::abs(tmp)*ats::abs(tmp);
+	  norm += ats::abs(b_host(k,i,l))*ats::abs(b_host(k,i,l));	
+	}
+      }
+      //printf("norm %e, residual %e, rel res %e\n", norm, residual, residual/norm);
```

Comment:
Minor format comment, this line can be removed; not necessary to change in the PR, just a note to cleanup later.